lib/devices.go

```diff
@@ -0,0 +1,238 @@
+package usb_tree_lib
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/google/gousb"
+	"github.com/google/gousb/usbid"
+)
+
+const (
+	StateNormal  LogState = "normal"
+	StateAdded   LogState = "added"
+	StateRemoved LogState = "removed"
+)
+
+type LogState string
```

Comment:
I would move this above the `const`s, otherwise reading top-down I have no idea what `LogState` is